Fixed #28194 -- Added support for normalization and cover density to SearchRank.

hannseman opened the pull request:
Ticket https://code.djangoproject.com/ticket/28194#ticket

charettes commented on docs/ref/contrib/postgres/search.txt:
```diff
@@ -137,6 +137,10 @@ order by relevancy::
+The ``cover_density`` parameter can be used to enable cover density ranking.
+This means that query words that are further apart from each other in the
+document will be penalized.
+
```
Missing a `::versionadded 3.1`

hannseman replied:
Forgot about that one. Thanks!

charettes commented on django/contrib/postgres/search.py:
```diff
@@ -208,7 +208,9 @@ class SearchRank(Func):
-    def __init__(self, vector, query, weights=None):
+    def __init__(self, vector, query, weights=None, cover_density=False):
+        if cover_density:
+            self.function = 'ts_rank_cd'
```
Minor but I'd move this control flow block after the `weights` one to match the args order.

hannseman replied:
Makes sense 👍 . Had a though that it might be good to set `self.function` as early as possible.

hannseman commented:
I added support for the `normalization` option per https://code.djangoproject.com/ticket/28194. I added it in a separate commit, let me know if I should squash them.

felixxm commented:
@hannseman Thanks :+1:  I added new verses to tests which show actual use cases, and pushed edits to docs.

hannseman commented:
@charettes @felixxm thanks for the review and merge!

@claudep thanks for finding the duplicate ticket!

🎉
